api/admin/controller.py

```diff
@@ -1600,26 +1604,35 @@ def custom_list(self, list_id):
             return MISSING_CUSTOM_LIST
 
         if flask.request.method == "GET":
-            entries = []
-            for entry in list.entries:
-                if entry.edition:
-                    url = self.url_for(
-                        "permalink",
-                        identifier_type=entry.edition.primary_identifier.type,
-                        identifier=entry.edition.primary_identifier.identifier,
-                        library_short_name=library.short_name,
-                    )
-                    entries.append(dict(identifier_urn=entry.edition.primary_identifier.urn,
-                                        title=entry.edition.title,
-                                        authors=[author.display_name for author in entry.edition.author_contributors],
-                                        medium=Edition.medium_to_additional_type.get(entry.edition.medium, None),
-                                        url=url,
-                                        language=entry.edition.language,
-                    ))
-            collections = []
-            for collection in list.collections:
-                collections.append(dict(id=collection.id, name=collection.name, protocol=collection.protocol))
-            return dict(id=list.id, name=list.name, entries=entries, collections=collections, entry_count=len(entries))
+            pagination = load_pagination_from_request()
+            if isinstance(pagination, ProblemDetail):
+                return pagination
+
+            all_works = self._db.query(Work).join(Work.custom_list_entries).filter(CustomListEntry.list_id==list_id)
+            page_of_works = pagination.apply(all_works)
+
+            worklist = WorkList()
+            worklist.initialize(library, customlists=[list])
+
+            annotator = self.manager.annotator(worklist)
+
+            url = self.cdn_url_for(
+                "crawlable_list_feed", list_name=list.name,
+                library_short_name=library.short_name,
+            )
+
+            feed = AcquisitionFeed(self._db, list.name, url, page_of_works, annotator)
```

Comment:
You can use `AcquisitionFeed.page` and pass in the pagination so you don't have to add the links yourself.

Reply:
I tried `AcquisitionFeed.page` initially but that has the issue that it queries against the materializedview so nothing gets returned to the feed.